eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -45,17 +48,34 @@ foreach ($branch in $branches)
   $openPullRequests = $pullRequests | ? { $_.State -eq "open" }
   if ($openPullRequests.Count -gt 0)
   {
-    LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has open pull Requests. Skipping"
+    LogDebug "Branch [ $branchName ] in repo [ $RepoId ] has open pull Requests. Skipping"
     LogDebug $openPullRequests.url
     continue
   }
 
-  LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has no associated open Pull Request. Deleting Branch"
-  try{
-    Remove-GitHubSourceReferences -RepoOwner $RepoOwner -RepoName $RepoName -Ref ($branch.Remove(0,5)) -AuthToken $AuthToken
+  if ($LastCommitOlderThan) {
+    if (!$res.object -or !$res.object.url) {
+      LogWarning "No commit url returned from response. Skipping... "
+      continue
+    }
+    try {
+      $commitDate = Get-GithubReferenceCommitDate -commitUrl $res.object.url
+      if ($commitDate -and ($commitDate -gt $LastCommitOlderThan)) {
+        LogDebug "The branch $branch last commit date $commitDate is newer than the date $LastCommitOlderThan. Skipping."
+        continue
+      }
+    }
+    catch {
+      LogError "Get-GithubReferenceCommitDate failed with exception:`n$_"
+      exit 1
+    }
+  } 
+  LogDebug "Branch [ $branchName ] in repo [ $RepoId ] has no associated open Pull Request. "
+  try {
+    Remove-GitHubSourceReferences -RepoId $RepoId -Ref ($branch -replace "refs/") -AuthToken $AuthToken
```

Comment:
```suggestion
    Remove-GitHubSourceReferences -RepoId $RepoId -Ref $branch -AuthToken $AuthToken
```
If we need to strip of the `refs/` it should be done inside of the Remove-GitHubSourceReferences function.

Reply:
There are three different branches probably passed to the function.
```
$branchName
heads/$branchName
refs/heads/$branchName
```

If we have the strip inside of the `Remove-GitHubSourceReferences`, then we only accept the last two branch format. If I have the $branchName only, then we need to add "heads/" in front. 

I don't think the Remove function is the place to take care of all these. It is better to align what github defined and force caller to pass the right thing.

Reply:
Talked offline. Agreed on having the trim inside of the function call.